lib/jekyll/excerpt.rb

```diff
@@ -155,7 +155,7 @@ def sanctify_liquid_tags(head)
       tag_names.flatten!
       tag_names.reverse_each do |tag_name|
         next unless liquid_block?(tag_name)
-        next if head =~ endtag_regex_stash(tag_name)
+        next if head&.match?(endtag_regex_stash(tag_name))
```

Comment:
Since `head` will only ever be `nil` or a String **and** that `endtag_regex_stash(tag_name)` will always be a regex, we can ditch the defensive coding and use `Regex#match?` instead:
```suggestion
        next if endtag_regex_stash(tag_name).match?(head)
```